lib/shared/layers/python-sdk/python/genai_core/user_feedback.py

```diff
@@ -0,0 +1,51 @@
+import os
+import uuid
+import boto3
+import json
+from pydantic import BaseModel
+from datetime import datetime
+
+dynamodb = boto3.resource("dynamodb")
+s3_client = boto3.client("s3")
+
+USER_FEEDBACK_BUCKET_NAME = os.environ.get("USER_FEEDBACK_BUCKET_NAME")
+
+
+def add_user_feedback(
+    sessionId: str,
+    key: str,
+    feedback: str,
+    prompt: str,
+    completion: str,
+    model: str,
+    userId: str
+):
+    feedbackId = str(uuid.uuid4())
+    timestamp = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+    
+    item = {
+        "feedbackId": feedbackId,
+        "sessionId": sessionId,
+        "userId": userId,
+        "key": key,
+        "prompt": prompt,
+        "completion": completion,
+        "model": model,
+        "feedback": feedback,
+        "createdAt": timestamp
+    }
+    
+    response = s3_client.put_object(
+        Bucket=USER_FEEDBACK_BUCKET_NAME,
+        Key=feedbackId,
+        Body=json.dumps(item),
+        ContentType="application/json",
+        StorageClass='STANDARD_IA',
```

Comment:
Let's handle storage class in CDK instead.

something along these line should ensure that objects are transitioned to the right class and it's easier to handle more complex rules in future

```
    const userFeedbackBucket = new s3.Bucket(this, 'UserFeedbackBucket', {
       ...
      lifecycleRules: [{
        transitions: [{
          storageClass: s3.StorageClass.INFREQUENT_ACCESS,
          transitionAfter: cdk.Duration.days(0), // Transition immediately after creation
        }],
      }],
    });
  }
```

Reply:
The minimum allowed value of transitionAfter days is 30. The recommended approach to directly store the objects in STANDARD_IA is setting the storage class at object level.